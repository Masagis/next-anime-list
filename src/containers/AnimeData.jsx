import React from 'react'

function AnimeData(props) {
  const { data, handleDetailAnime } = props

  return (
    <div
      data-testid="card-list-anime"
      aria-hidden
      className="col-md-6"
      style={{ cursor: 'pointer' }}
      onClick={() => handleDetailAnime()}
    >
      <div className="row">
        <div className="col-xs-6 col-sm-6 col-md-3 col-6">
          <img
            data-testid="image-list-anime"
            src={`${data?.images?.jpg?.image_url}`}
            alt="banner-anime"
            className="mb-2 img-fluid rounded"
            width="150px"
            height="150px"
          />
        </div>
        <div className="col-xs-6 col-sm-6 col-md-9 col-6">
          <div>
            <h4 data-testid="title-list-anime">{data?.title}</h4>
            <p data-testid="type-list-anime">
              Type :{' '}
              <span
                className={`badge ${
                  data?.type === 'TV'
                    ? 'bg-warning'
                    : data?.type === 'Movie'
                    ? 'bg-info'
                    : 'bg-secondary'
                }`}
              >
                {data?.type}
              </span>
            </p>
            <p data-testid="rating-list-anime">Rating : {data?.score}</p>
            <p data-testid="rilis-list-anime">
              Released on : {data?.aired?.string}
            </p>
          </div>
        </div>
      </div>
    </div>
  )
}

export default AnimeData
