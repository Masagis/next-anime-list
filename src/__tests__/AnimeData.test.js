/* eslint-disable no-undef */
import { fireEvent, render, screen } from '@testing-library/react'
import AnimeData from '../containers/AnimeData'

describe('Component AnimeDataList', () => {
  test('should be render the component card anime dta list', () => {
    const data = {
      mal_id: 1,
      url: 'https://myanimelist.net/anime/1/Cowboy_Bebop',
      images: {
        jpg: {
          image_url: 'https://cdn.myanimelist.net/images/anime/4/19644.jpg',
          small_image_url:
            'https://cdn.myanimelist.net/images/anime/4/19644t.jpg',
          large_image_url:
            'https://cdn.myanimelist.net/images/anime/4/19644l.jpg',
        },
        webp: {
          image_url: 'https://cdn.myanimelist.net/images/anime/4/19644.webp',
          small_image_url:
            'https://cdn.myanimelist.net/images/anime/4/19644t.webp',
          large_image_url:
            'https://cdn.myanimelist.net/images/anime/4/19644l.webp',
        },
      },

      title: 'Cowboy Bebop',
      title_english: 'Cowboy Bebop',
      title_japanese: 'カウボーイビバップ',
      type: 'TV',
      source: 'Original',
      episodes: 26,
      status: 'Finished Airing',
      airing: false,
      aired: {
        from: '1998-04-03T00:00:00+00:00',
        to: '1999-04-24T00:00:00+00:00',
        prop: {
          from: {
            day: 3,
            month: 4,
            year: 1998,
          },
          to: {
            day: 24,
            month: 4,
            year: 1999,
          },
        },
        string: 'Apr 3, 1998 to Apr 24, 1999',
      },
      duration: '24 min per ep',
      rating: 'R - 17+ (violence & profanity)',
      score: 8.75,
      scored_by: 859099,
    }

    const handleDetailAnime = jest.fn()

    render(<AnimeData data={data} handleDetailAnime={handleDetailAnime} />)
    expect(screen.getByTestId('card-list-anime')).toBeInTheDocument()
    expect(screen.getByTestId('image-list-anime')).toHaveAttribute(
      'src',
      'https://cdn.myanimelist.net/images/anime/4/19644.jpg',
    )
    expect(screen.getByTestId('title-list-anime')).toHaveTextContent(
      'Cowboy Bebop',
    )
    expect(screen.getByTestId('type-list-anime')).toHaveTextContent('TV')
    expect(screen.getByTestId('rating-list-anime')).toHaveTextContent(8.75)
    expect(screen.getByTestId('rilis-list-anime')).toHaveTextContent(
      'Apr 3, 1998 to Apr 24, 1999',
    )

    fireEvent.click(screen.getByTestId('card-list-anime'))
    expect(handleDetailAnime).toBeCalledTimes(1)
  })

  test('snapshoot', () => {
    const data = {
      mal_id: 1,
      url: 'https://myanimelist.net/anime/1/Cowboy_Bebop',
      images: {
        jpg: {
          image_url: 'https://cdn.myanimelist.net/images/anime/4/19644.jpg',
          small_image_url:
            'https://cdn.myanimelist.net/images/anime/4/19644t.jpg',
          large_image_url:
            'https://cdn.myanimelist.net/images/anime/4/19644l.jpg',
        },
        webp: {
          image_url: 'https://cdn.myanimelist.net/images/anime/4/19644.webp',
          small_image_url:
            'https://cdn.myanimelist.net/images/anime/4/19644t.webp',
          large_image_url:
            'https://cdn.myanimelist.net/images/anime/4/19644l.webp',
        },
      },

      title: 'Cowboy Bebop',
      title_english: 'Cowboy Bebop',
      title_japanese: 'カウボーイビバップ',
      type: 'TV',
      source: 'Original',
      episodes: 26,
      status: 'Finished Airing',
      airing: false,
      aired: {
        from: '1998-04-03T00:00:00+00:00',
        to: '1999-04-24T00:00:00+00:00',
        prop: {
          from: {
            day: 3,
            month: 4,
            year: 1998,
          },
          to: {
            day: 24,
            month: 4,
            year: 1999,
          },
        },
        string: 'Apr 3, 1998 to Apr 24, 1999',
      },
      duration: '24 min per ep',
      rating: 'R - 17+ (violence & profanity)',
      score: 8.75,
      scored_by: 859099,
    }

    const renderComponent = render(<AnimeData data={data} />)
    expect(renderComponent).toMatchSnapshot()
  })
})
