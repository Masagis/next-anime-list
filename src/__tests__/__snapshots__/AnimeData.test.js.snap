// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component AnimeDataList snapshoot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-hidden="true"
        class="col-md-6"
        data-testid="card-list-anime"
        style="cursor: pointer;"
      >
        <div
          class="row"
        >
          <div
            class="col-xs-6 col-sm-6 col-md-3 col-6"
          >
            <img
              alt="banner-anime"
              class="mb-2 img-fluid rounded"
              data-testid="image-list-anime"
              height="150px"
              src="https://cdn.myanimelist.net/images/anime/4/19644.jpg"
              width="150px"
            />
          </div>
          <div
            class="col-xs-6 col-sm-6 col-md-9 col-6"
          >
            <div>
              <h4
                data-testid="title-list-anime"
              >
                Cowboy Bebop
              </h4>
              <p
                data-testid="type-list-anime"
              >
                Type :
                 
                <span
                  class="badge bg-warning"
                >
                  TV
                </span>
              </p>
              <p
                data-testid="rating-list-anime"
              >
                Rating : 
                8.75
              </p>
              <p
                data-testid="rilis-list-anime"
              >
                Released on : 
                Apr 3, 1998 to Apr 24, 1999
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-hidden="true"
      class="col-md-6"
      data-testid="card-list-anime"
      style="cursor: pointer;"
    >
      <div
        class="row"
      >
        <div
          class="col-xs-6 col-sm-6 col-md-3 col-6"
        >
          <img
            alt="banner-anime"
            class="mb-2 img-fluid rounded"
            data-testid="image-list-anime"
            height="150px"
            src="https://cdn.myanimelist.net/images/anime/4/19644.jpg"
            width="150px"
          />
        </div>
        <div
          class="col-xs-6 col-sm-6 col-md-9 col-6"
        >
          <div>
            <h4
              data-testid="title-list-anime"
            >
              Cowboy Bebop
            </h4>
            <p
              data-testid="type-list-anime"
            >
              Type :
               
              <span
                class="badge bg-warning"
              >
                TV
              </span>
            </p>
            <p
              data-testid="rating-list-anime"
            >
              Rating : 
              8.75
            </p>
            <p
              data-testid="rilis-list-anime"
            >
              Released on : 
              Apr 3, 1998 to Apr 24, 1999
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
